woocommerce/woocommerce-ios
# Order Details > Shipping Label: present shipping label printing instructions screen

Fixes #2174 

## Why

For users who haven't printed from an iOS device, we show a screen with instructions on how to do so with 4 steps.

## Changes

Note: please check p91TBi-3V3-p2 for us to discuss on SwiftUI adoption.

- Added two assets for `woo-shipping-label-printing-instructions` in light and dark mode
- Created 3 SwiftUI views:
  - `BoldableTextView`: this view shows a boldable string (as shown in screenshots), and has the most logic of all views in this PR. I looked into how to show text with different weights, and [this blog post](https://www.avanderlee.com/swiftui/text-weight-combinations/) best summarizes my findings and my implementation is based on the solution in the blog post. The solution in the blog post has some bugs handling space, and I changed and cleaned up quite a bit. I also added unit tests for different string content scenarios
  - `ShippingLabelPrintingStepView`: displays one step of printing instructions given its `ViewModel`
  - `ShippingLabelPrintingInstructionsView`: the top-level view that consists of an illustration at the top and four steps (`ShippingLabelPrintingStepView`). the previews show the view in light and dark mode
- In `OrderDetailsViewModel`, presented `ShippingLabelPrintingInstructionsViewController` when tapping on the shipping label printing info row

## Testing

Prerequisites: the site has [WooCommerce Shipping & Tax plugin](https://docs.woocommerce.com/document/woocommerce-shipping-and-tax/) and there is an order with at least one non-refunded shipping label.

You can set up your store to use a test credit card with instructions in p91TBi-3xD-p2.

- Go to the orders tab
- Tap on an order with at least one non-refunded shipping label
- After a shipping label card shows up, tap on the info row "Don't know how to print from your phone?" --> a screen should be presented that displays 4 steps on how to print a shipping label from an iOS device
- Tap on the close button ("X") --> the instructions screen should be dismissed

## Example screenshots

\ | dark | light
-- | -- | --
iPhone 11 - portrait | ![Simulator Screen Shot - iPhone 11 - 2020-12-04 at 11 39 00](https://user-images.githubusercontent.com/1945542/101119726-ee52e800-3626-11eb-9d8f-efa226a50da3.png) | ![Simulator Screen Shot - iPhone 11 - 2020-12-04 at 11 39 14](https://user-images.githubusercontent.com/1945542/101119736-f27f0580-3626-11eb-8c2e-b80ae271a53d.png)
iPhone 11 - landscape | ![Simulator Screen Shot - iPhone 11 - 2020-12-04 at 11 39 04](https://user-images.githubusercontent.com/1945542/101119764-01fe4e80-3627-11eb-8cb9-73ad97e58ddd.png) | ![Simulator Screen Shot - iPhone 11 - 2020-12-04 at 11 39 19](https://user-images.githubusercontent.com/1945542/101119775-062a6c00-3627-11eb-9023-87c54be692e4.png)
iPad Air gen 3 | ![Simulator Screen Shot - iPad Air (3rd generation) - 2020-12-04 at 11 41 30](https://user-images.githubusercontent.com/1945542/101119847-307c2980-3627-11eb-8420-d6b63e9436b7.png) | ![Simulator Screen Shot - iPad Air (3rd generation) - 2020-12-04 at 11 41 58](https://user-images.githubusercontent.com/1945542/101119856-35d97400-3627-11eb-9278-7257adbc8c53.png)





Update release notes:
